### ISSUE TITLE
Is this project looking for contributions?

### ISSUE THREAD

**Issue by malespiaut** [wdlkmpx/gwaveedit#1]
Hello,

I've been working on mhWaveEdit for a few days, before realizing that forks already existed, with the same goal of getting rid of the deprecated GTK2 symbols, in the hope of porting the software to GTK3, and possibly GTK4.

As a lot of smart work has already been done, I would like to know if you are interested to reboot the project, because I have some potential pull requests, and it's more motivating to be two on a project.

Best regards.

**Comment by wdlkmpx** [comment]
I ported a few small apps to gtk3 while keeping gtk2 but this one requires some advanced work and a lot of time , the gobject and Cairo stuff is not as easy as it seems

I won't get rid of gtk2, but I can accept contributions. I'll replace autotools with my own fast configure script which is faster than cmake, meson, etc

**Comment by wdlkmpx** [comment]
Gtk4 is a whole different beast, they removed quite a lot of widgets, so it requires a major rewrite, I'm not interested in gtk4 right now

Some of the GtkObject code will break if you just replace it with  GObject, gtk_object_ref/sink stuff, some research is required to see where things fail with gobject. I replaced the whole codebase with GObject, but I had to revert the change, but it can done slowly, one file at a time

The first thing that must be done is apply proper indentation, there is mixed space and tab indentation, the code is unreadable, ideally the proper indentation should be 4 spaces

So making this at least compile with gtk3 is a major task, I accept gtk3 contributions and I guess I can add ifdefs to make the gtk2 port compile again, it's unlikely the gtk3 stuff will be stable

**Comment by wdlkmpx** [comment]
In other branches you can see some experimental work that I effectively abandoned a long time ago, wow, time passes by so fast man
https://github.com/wdlkmpx/gwaveedit/branches

**Comment by wdlkmpx** [comment]
Just take your time to see what you can improve, and make sure to perform tests as much as you can. Your cmake stuff will be accepted of course, one of the 2 build systems for da project, it's ok for the cmake stuff to not support gtk2

**Comment by malespiaut** [comment]
Hello,

Thank you for your answer. From what I understand it seems that this project have already been pushed to the maximum.

I'll see if I can come up with anything interesting.

Best regards.

**Comment by wdlkmpx** [comment]
Yeah an insane amount of changes have happened, I also added SDL2 support, gwaveedit supports SDL & SDL2 audio output

@gkarsay actually did a lot of work, so when I continued the effort, I didn't realize another insane amount of work was needed

Changing 20,000+ lines of code is basically the standard procedure when it comes to updating many gtk apps, my stats show similar numbers in all the projects

But in the end, some advanced cairo stuff is needed, I don't think many people understand that stuff to replace the Gdk code, I ported a few small games to GTK3, and the Cairo stuff was the most complex stuff to implement, gwaveedit is even more complex than those tetris games, and the code is unreadable and convoluted, needs a major cleanup and consistent indentation

I see this more like a technical challenge, I mostly use audacity, but indeed, most of what can be done by mere mortals has already been done. It's possible to add pure gtk3 stuff, but the app won't work without the pure C.a.i.ro implementation of the Wave

**Comment by wdlkmpx** [comment]
LOL I just realized the app is apparently broken, I'll try to fix it
Nobody uses this version obviously

**Comment by wdlkmpx** [comment]
I fixed the application, it was broken for more than a year ![aaaa](https://user-images.githubusercontent.com/10052849/155864311-bc222175-6efa-4a2a-a6f2-6884fa203b40.jpg)


new contributions are welcome

**Comment by mckaygerhard** [comment]
hi @wdlkmpx  i asked about status of several project from you..  but no response..  specially cdimage2iso

**Comment by wdlkmpx** [comment]
They are all unmaintaned,  I must work to make ends meet
